lib/validator/v3/validator.go

```diff
@@ -828,6 +829,23 @@ func validateNodeSpec(v *validator.Validate, structLevel *validator.StructLevel)
 	}
 }
 
+func validateBGPPeerSpec(v *validator.Validate, structLevel *validator.StructLevel) {
```

Comment:
This looks good - I think we should add one bit to the validation of the node spec as well for the RR cluster ID.

IIUC, it needs to be a valid IPv4 address.